qiskit/qpy/common.py

```diff
@@ -18,7 +18,7 @@
 import io
 import struct
 
-from qiskit.qpy import formats
+from qiskit.qpy import formats, type_keys
 
 QPY_VERSION = 7
```

Comment:
Should QPY version be bumped?

Reply:
I was not sure. I would like to hear from a QPY expert.

Reply:
This guidance was a little surprising to me at first since we are adding a new data type, but I think it makes sense. Here is how I think about it that makes sense for me:

qpy attempts to provide forwards compatibility for loading previously stored circuits (I think there is still freedom for terra to completely remove something and then it is okay not to load that any more). Since the way data is stored for a specific object might need to change, qpy versions its files and can use the version number to know to load an object stored with the old format. Here we are adding new types to be stored, so the change has no impact on loading previously stored data. The part that was confusing to me is that we could store a kernel in terra 0.25 with qpy version 7 and terra 0.24 would also have qpy version 7 but not be able to load the kernel. However, the terra version is also stored in the qpy data and a separate warning is generated when trying to load qpy data with an earlier version of terra than the one used to generate the data, so I think that covers that case.